crates/fluvio-sc/src/controllers/mirroring/controller.rs

```diff
@@ -0,0 +1,220 @@
+use std::time::{Duration, SystemTime};
+use fluvio_controlplane_metadata::mirroring::{MirrorConnect, MirroringRemoteClusterRequest};
+
+use fluvio_socket::{ClientConfig, MultiplexerSocket};
+use futures_util::StreamExt;
+use fluvio_future::{task::spawn, timer::sleep};
+use fluvio_sc_schema::{
+    core::MetadataItem,
+    mirror::{ConnectionStatus, MirrorPairStatus, MirrorSpec, MirrorStatus, MirrorType},
+    mirroring::ObjectMirroringRequest,
+    topic::{MirrorConfig, RemoteMirrorConfig, ReplicaSpec, TopicSpec},
+    TryEncodableFrom,
+};
+use fluvio_stream_dispatcher::store::StoreContext;
+use tracing::{debug, error, info, instrument};
+use anyhow::{anyhow, Context, Result};
+
+use crate::core::SharedContext;
+
+const MIRRORING_CONTROLLER_INTERVAL: u64 = 5;
+const MIRRORING_CONTROLLER_RETRY_INTERVAL: u64 = 10;
+
+// This is the main controller for the mirroring feature.
+// Remote Clusters will connect to the home clusters.
+pub struct MirroringController<C: MetadataItem> {
```

Comment:
Renamed as `RemoteMirrorController` or something like that.  That is this is controller for Remote

Reply:
Fixed